### PR TITLE
Test/e2e like

### PR DESCRIPTION
**Pull request by 276746**
# E2E test: Like Logic:
## Test flow:
- Sign In
- Like an Itinerary in the OverviewScreen
- Go to LikedScreen
- Check that the Itinerary is liked
- Unlike the Itinerary
- Go back to OverviewScreen
- Go back to LikedScreen
- Check if there is no liked Itinerary

Special thanks to @ethangraham2001 who fixed a bug with protobuf in commit: 
- [Fixed weird dependency issue with protobufs that was breaking](https://github.com/WanderWise-Inc/app/pull/277/commits/c7343f744e38b6e5d206a4272edf5c94b161f33b)

### CODE REVIEW

File: app/build.gradle.kts [hunk 1]
```diff
@@ -240,6 +238,14 @@ dependencies {
...
 }
 
+// fixes androidTests not running via dark magic
+configurations.all {
+    resolutionStrategy.dependencySubstitution {
+        substitute(module("com.google.protobuf:protobuf-lite"))
+            .using(module("com.google.protobuf:protobuf-javalite:3.23.0"))
+    }
+}
+
```
Comment (ethangraham2001) [inline app/build.gradle.kts:248]: Brilliant addition

File: app/src/androidTest/java/com/github/wanderwise_inc/app/data/E2EProfileRepository.kt [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+  override fun removeItineraryFromLiked(userUid: String, itineraryUid: String) {
+    repository.find { it.userUid == userUid }.also { it?.likedItinerariesUid?.remove(itineraryUid) }
+  }
+
+  override suspend fun checkIfItineraryIsLiked(userUid: String, itineraryUid: String): Boolean {
+    repository
+        .find { it.userUid == userUid }
+        .also {
+          it?.let { profile ->
```
Comment (ethangraham2001) [inline app/src/androidTest/java/com/github/wanderwise_inc/app/data/E2EProfileRepository.kt:45]: homie used a `?.let` :o

File: app/src/androidTest/java/com/github/wanderwise_inc/app/di/AppModule.kt
Comment (ethangraham2001) [inline app/src/androidTest/java/com/github/wanderwise_inc/app/di/AppModule.kt:1]: Looks good

File: app/src/androidTest/java/com/github/wanderwise_inc/app/e2e/LikeE2ETest.kt
Comment (ethangraham2001) [inline app/src/androidTest/java/com/github/wanderwise_inc/app/e2e/LikeE2ETest.kt:1]: I've not got any experience with E2E tests, but this looks clean. Good job

**Review by ethangraham2001 — APPROVED** [review]
Good work

File: app/build.gradle.kts (hunk 1 above)
Comment (engjellismaili) [reply app/build.gradle.kts:248]: What is the purpose of this addition ?

Comment (276746) [reply app/build.gradle.kts:248]: Only a genius would have done it

File: app/src/androidTest/java/com/github/wanderwise_inc/app/data/E2EDirectionsRepository.kt [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import com.google.android.gms.maps.model.LatLng
+
+class E2EDirectionsRepository : DirectionsRepository {
+  override fun getPolylineWayPoints(
+      origin: String,
+      destination: String,
+      waypoints: List<String>,
+      apiKey: String
+  ): LiveData<List<LatLng>?> {
+    TODO("Not yet implemented")
```
Comment (engjellismaili) [inline app/src/androidTest/java/com/github/wanderwise_inc/app/data/E2EDirectionsRepository.kt:13]: The TODO should be removed before merging into main. Is there a way to modify it or it is necessary to leave it like that ?

File: app/src/androidTest/java/com/github/wanderwise_inc/app/data/E2EImageRepository.kt [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  }
+
+  override fun setIsItineraryImage(isItineraryImage: Boolean) {
+    TODO("Not yet implemented")
+  }
+
+  override fun deleteImageFromStorage(fileName: String) {
+    TODO("Not yet implemented")
+  }
+}
```
Comment (engjellismaili) [inline app/src/androidTest/java/com/github/wanderwise_inc/app/data/E2EImageRepository.kt:44]: Same comment as before

File: app/src/androidTest/java/com/github/wanderwise_inc/app/data/E2EProfileRepository.kt (hunk 2 above)
Comment (engjellismaili) [reply app/src/androidTest/java/com/github/wanderwise_inc/app/data/E2EProfileRepository.kt:45]: do you prefer to use an if else ? using let is cleaner ⚔️

File: app/src/androidTest/java/com/github/wanderwise_inc/app/data/E2EProfileRepository.kt
Comment (engjellismaili) [inline app/src/androidTest/java/com/github/wanderwise_inc/app/data/E2EProfileRepository.kt:1]: Lgtm

File: app/src/androidTest/java/com/github/wanderwise_inc/app/di/AppModule.kt
Comment (engjellismaili) [inline app/src/androidTest/java/com/github/wanderwise_inc/app/di/AppModule.kt:1]: Clean code, are you a senior developper by any chance ?

File: app/src/androidTest/java/com/github/wanderwise_inc/app/e2e/LikeE2ETest.kt
Comment (engjellismaili) [inline app/src/androidTest/java/com/github/wanderwise_inc/app/e2e/LikeE2ETest.kt:1]: Looks good, to me, do you have the coverage of the tests ?

File: app/src/main/java/com/github/wanderwise_inc/app/di/AppModule.kt
Comment (engjellismaili) [inline app/src/main/java/com/github/wanderwise_inc/app/di/AppModule.kt:1]: Is there a reason to specify types ?

File: app/src/main/java/com/github/wanderwise_inc/app/ui/navigation/NavigationActions.kt [hunk 5]
```diff
@@ -26,80 +27,100 @@ object CreationPreviewOptions {
   const val PREVIEW_ITINERARY = "Itinerary"
 }
 
-sealed class Destination(val route: String, val icon: Int, @StringRes val textId: Int) {
-  sealed class TopLevelDestination(route: String, icon: Int, textId: Int) :
-      Destination(route, icon, textId) {
+sealed class Destination(
+    val route: String,
+    val icon: Int,
+    @StringRes val textId: Int,
```
Comment (engjellismaili) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/navigation/NavigationActions.kt:33]: What does @StringRes ?

**Review by engjellismaili — APPROVED** [review]
Overall clean code and good testing. Making E2E tests looks hard and is really important for the app. Well done

**Pull request merged**